tests/DnaSequenceTest.php

```diff
@@ -0,0 +1,41 @@
+<?php declare(strict_types=1);
+
+namespace MLL\Utils\Tests;
+
+use MLL\Utils\DnaSequence;
+use PHPUnit\Framework\TestCase;
+
+class DnaSequenceTest extends TestCase
+{
+    public function testReverse(): void
+    {
+        $dnaSequence = new DnaSequence('ATGC');
+        self::assertEquals('CGTA', $dnaSequence->reverse());
+    }
+
+    public function testComplement(): void
+    {
+        $dnaSequence = new DnaSequence('ATGC');
+        self::assertEquals('TACG', $dnaSequence->complement());
+    }
+
+    public function testReverseComplement(): void
+    {
+        $dnaSequence = new DnaSequence('ATGC');
+        self::assertEquals('GCAT', $dnaSequence->reverseComplement());
+    }
+
+    public function testInvalidSequenceThrowsException(): void
+    {
+        $this->expectException(\InvalidArgumentException::class);
+        new DnaSequence('INVALID');
+    }
+
+    public function testEmptySequence(): void
+    {
+        $dnaSequence = new DnaSequence('');
```

Comment:
Duerfen hier ueberhaupt leere DNA Sequenzen vorkommen?

Reply:
Ist ein bisschen eine Frage des Standpunktes. Ich stelle mir aber gerade eine Situation vor in der man den "diff" von zwei Sequencen berechnet, der ja auch wieder eine Sequenz wäre. In dem Fall wäre es relativ umständlich immer die Exception abzufangen, dass ein leerer String keine valide Sequenz ist. 
Daher würde ich es wohl erstmal so lassen..